### ISSUE TITLE
Unnecessary newline after long generics in function

### ISSUE THREAD

**Issue by nrc** [rust-lang-nursery/rustfmt#548]
```
-    pub fn build_texture_atlas<'a, T: IntoIterator<Item = &'a WadName>>(
-            &'a self, names_iter: T) -> (TransparentImage, BoundsLookup) {
-        let entries = ordered_atlas_entries(&self.animated_walls,
-                                            |n| self.texture(&n),
-                                            names_iter);
+    pub fn build_texture_atlas<'a, T: IntoIterator<Item = &'a WadName>>
+                                                                        (&'a self,
+                                                                         names_iter: T)
+                                                                         -> (TransparentImage, BoundsLookup) {
```

**Comment by nrc** [comment]
Would be nice to automatically insert a where clause here.

**Comment by nrc** [comment]
Fixed by #666
